pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed

pkg/ccl/serverccl/statusccl/statusccl_test.TestTenantStatusAPI [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20344192?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20344192?buildTab=artifacts#/) on master @ [1bedd2556578167dff1349b3ed641cc622b1e6a0](https://github.com/cockroachdb/cockroach/commits/1bedd2556578167dff1349b3ed641cc622b1e6a0):


```
=== RUN   TestTenantStatusAPI
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/7067ad80d000d85ce2eeee021e911c58/logTestTenantStatusAPI1189584711
    tenant_status_test.go:149: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/7067ad80d000d85ce2eeee021e911c58/logTestTenantStatusAPI1189584711
--- FAIL: TestTenantStatusAPI (10.98s)
=== RUN   TestTenantStatusAPI/reset_sql_stats
--- FAIL: TestTenantStatusAPI/reset_sql_stats (3.40s)
=== RUN   TestTenantStatusAPI/reset_sql_stats/flushed=true
    tenant_status_test.go:687: 
        	Error Trace:	pkg/ccl/serverccl/statusccl/tenant_status_test.go:975
        	            				pkg/ccl/serverccl/statusccl/tenant_status_test.go:687
        	Error:      	Should be true
        	Test:       	TestTenantStatusAPI/reset_sql_stats/flushed=true
        	Messages:   	expected SELECT _, _, _ to be found in control tenant cluster, but it was not found
--- FAIL: TestTenantStatusAPI/reset_sql_stats/flushed=true (1.70s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStatusAPI.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/serverccl/statusccl/statusccl_test.TestTenantStatusAPI [failed](https://mesolite.cluster.engflow.com/invocations/default/bd0a7c64-fbf9-4544-ada1-73c53840e19b?testReportRun=6&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3NlcnZlcmNjbC9zdGF0dXNjY2w6c3RhdHVzY2NsX3Rlc3Q=) on master @ [364231907b129337ae16fc2e1f93c5defaaba395](https://github.com/cockroachdb/cockroach/commits/364231907b129337ae16fc2e1f93c5defaaba395):


```
=== RUN   TestTenantStatusAPI
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestTenantStatusAPI864146763
    tenant_status_test.go:149: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantStatusAPI864146763
--- FAIL: TestTenantStatusAPI (13.43s)
=== RUN   TestTenantStatusAPI/reset_sql_stats
--- FAIL: TestTenantStatusAPI/reset_sql_stats (3.98s)
=== RUN   TestTenantStatusAPI/reset_sql_stats/flushed=false
    tenant_status_test.go:650: 
        	Error Trace:	pkg/ccl/serverccl/statusccl/tenant_status_test.go:975
        	            				pkg/ccl/serverccl/statusccl/tenant_status_test.go:650
        	Error:      	Should be true
        	Test:       	TestTenantStatusAPI/reset_sql_stats/flushed=false
        	Messages:   	expected SELECT _, _, _ to be found in test tenant cluster, but it was not found
--- FAIL: TestTenantStatusAPI/reset_sql_stats/flushed=false (1.76s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=6</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStatusAPI.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/serverccl/statusccl/statusccl_test.TestTenantStatusAPI [failed](https://mesolite.cluster.engflow.com/invocations/default/d87221bd-834a-4b6c-a705-60d8d57f0d69?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3NlcnZlcmNjbC9zdGF0dXNjY2w6c3RhdHVzY2NsX3Rlc3Q=) on master @ [fbc73a7cc20ef29891787ff5c3b30c36b150f908](https://github.com/cockroachdb/cockroach/commits/fbc73a7cc20ef29891787ff5c3b30c36b150f908):


```
=== RUN   TestTenantStatusAPI
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestTenantStatusAPI2451028638
    tenant_status_test.go:149: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantStatusAPI2451028638
--- FAIL: TestTenantStatusAPI (13.92s)
=== RUN   TestTenantStatusAPI/reset_sql_stats
--- FAIL: TestTenantStatusAPI/reset_sql_stats (3.43s)
=== RUN   TestTenantStatusAPI/reset_sql_stats/flushed=false
    tenant_status_test.go:650: 
        	Error Trace:	pkg/ccl/serverccl/statusccl/tenant_status_test.go:975
        	            				pkg/ccl/serverccl/statusccl/tenant_status_test.go:650
        	Error:      	Should be true
        	Test:       	TestTenantStatusAPI/reset_sql_stats/flushed=false
        	Messages:   	expected SELECT _, _, _ to be found in test tenant cluster, but it was not found
--- FAIL: TestTenantStatusAPI/reset_sql_stats/flushed=false (1.20s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStatusAPI.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/serverccl/statusccl/statusccl_test.TestTenantStatusAPI [failed](https://mesolite.cluster.engflow.com/invocations/default/70ddfa3d-db44-4c79-9082-d8cff4f711a9?testReportRun=8&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3NlcnZlcmNjbC9zdGF0dXNjY2w6c3RhdHVzY2NsX3Rlc3Q=) on master @ [e3770b8d2034d7eeaa5095a3b2214da3f35b895e](https://github.com/cockroachdb/cockroach/commits/e3770b8d2034d7eeaa5095a3b2214da3f35b895e):


```
=== RUN   TestTenantStatusAPI
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestTenantStatusAPI161025622
    tenant_status_test.go:149: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantStatusAPI161025622
--- FAIL: TestTenantStatusAPI (13.00s)
=== RUN   TestTenantStatusAPI/reset_sql_stats
--- FAIL: TestTenantStatusAPI/reset_sql_stats (3.94s)
=== RUN   TestTenantStatusAPI/reset_sql_stats/flushed=false
    tenant_status_test.go:650: 
        	Error Trace:	pkg/ccl/serverccl/statusccl/tenant_status_test.go:975
        	            				pkg/ccl/serverccl/statusccl/tenant_status_test.go:650
        	Error:      	Should be true
        	Test:       	TestTenantStatusAPI/reset_sql_stats/flushed=false
        	Messages:   	expected SELECT _, _, _ to be found in test tenant cluster, but it was not found
--- FAIL: TestTenantStatusAPI/reset_sql_stats/flushed=false (1.38s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=8</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStatusAPI.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/serverccl/statusccl/statusccl_test.TestTenantStatusAPI [failed](https://mesolite.cluster.engflow.com/invocations/default/035861b7-281d-4598-9417-ee3d52ff3518?testReportRun=9&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3NlcnZlcmNjbC9zdGF0dXNjY2w6c3RhdHVzY2NsX3Rlc3Q=) on master @ [ee0bb8fa3912df192238a61e6d0e471f14408b2c](https://github.com/cockroachdb/cockroach/commits/ee0bb8fa3912df192238a61e6d0e471f14408b2c):


```
=== RUN   TestTenantStatusAPI
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestTenantStatusAPI3895511153
    tenant_status_test.go:149: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantStatusAPI3895511153
--- FAIL: TestTenantStatusAPI (12.91s)
=== RUN   TestTenantStatusAPI/reset_sql_stats
--- FAIL: TestTenantStatusAPI/reset_sql_stats (3.56s)
=== RUN   TestTenantStatusAPI/reset_sql_stats/flushed=false
    tenant_status_test.go:650: 
        	Error Trace:	pkg/ccl/serverccl/statusccl/tenant_status_test.go:975
        	            				pkg/ccl/serverccl/statusccl/tenant_status_test.go:650
        	Error:      	Should be true
        	Test:       	TestTenantStatusAPI/reset_sql_stats/flushed=false
        	Messages:   	expected SELECT _, _, _ to be found in test tenant cluster, but it was not found
--- FAIL: TestTenantStatusAPI/reset_sql_stats/flushed=false (1.38s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=9</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStatusAPI.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/serverccl/statusccl/statusccl_test.TestTenantStatusAPI [failed](https://mesolite.cluster.engflow.com/invocations/default/bc01c76d-80ff-4af7-84ec-8d682f308a3c?testReportRun=9&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3NlcnZlcmNjbC9zdGF0dXNjY2w6c3RhdHVzY2NsX3Rlc3Q=) on master @ [9a77ab62090722887cc70dc1d33c1345a2d675a6](https://github.com/cockroachdb/cockroach/commits/9a77ab62090722887cc70dc1d33c1345a2d675a6):


```
=== RUN   TestTenantStatusAPI
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestTenantStatusAPI933615381
    tenant_status_test.go:149: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantStatusAPI933615381
--- FAIL: TestTenantStatusAPI (12.68s)
=== RUN   TestTenantStatusAPI/reset_sql_stats
--- FAIL: TestTenantStatusAPI/reset_sql_stats (3.67s)
=== RUN   TestTenantStatusAPI/reset_sql_stats/flushed=false
    tenant_status_test.go:650: 
        	Error Trace:	pkg/ccl/serverccl/statusccl/tenant_status_test.go:975
        	            				pkg/ccl/serverccl/statusccl/tenant_status_test.go:650
        	Error:      	Should be true
        	Test:       	TestTenantStatusAPI/reset_sql_stats/flushed=false
        	Messages:   	expected SELECT _, _, _ to be found in test tenant cluster, but it was not found
--- FAIL: TestTenantStatusAPI/reset_sql_stats/flushed=false (1.53s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=9</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStatusAPI.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/serverccl/statusccl/statusccl_test.TestTenantStatusAPI [failed](https://mesolite.cluster.engflow.com/invocations/default/9921d362-35f3-4937-a6a7-b8e427246718?testReportRun=10&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3NlcnZlcmNjbC9zdGF0dXNjY2w6c3RhdHVzY2NsX3Rlc3Q=) on master @ [fa76ba12e60927d8aa5ebad0eb64221f121927de](https://github.com/cockroachdb/cockroach/commits/fa76ba12e60927d8aa5ebad0eb64221f121927de):


```
=== RUN   TestTenantStatusAPI
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestTenantStatusAPI2529756866
    tenant_status_test.go:149: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantStatusAPI2529756866
--- FAIL: TestTenantStatusAPI (12.42s)
=== RUN   TestTenantStatusAPI/reset_sql_stats
--- FAIL: TestTenantStatusAPI/reset_sql_stats (3.48s)
=== RUN   TestTenantStatusAPI/reset_sql_stats/flushed=true
    tenant_status_test.go:687: 
        	Error Trace:	pkg/ccl/serverccl/statusccl/tenant_status_test.go:975
        	            				pkg/ccl/serverccl/statusccl/tenant_status_test.go:687
        	Error:      	Should be true
        	Test:       	TestTenantStatusAPI/reset_sql_stats/flushed=true
        	Messages:   	expected SELECT _, _, _ to be found in control tenant cluster, but it was not found
--- FAIL: TestTenantStatusAPI/reset_sql_stats/flushed=true (1.49s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=10</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStatusAPI.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ccl/serverccl/statusccl/tenant_status_test.go
+++ b/pkg/ccl/serverccl/statusccl/tenant_status_test.go
@@ -57,7 +57,11 @@ func TestTenantStatusAPI(t *testing.T) {
 	ctx := context.Background()
 
 	var knobs base.TestingKnobs
-	knobs.SQLStatsKnobs = sqlstats.CreateTestingKnobs()
+
+	sqlStatsKnobs := sqlstats.CreateTestingKnobs()
+	sqlStatsKnobs.SynchronousSQLStats = true
+
+	knobs.SQLStatsKnobs = sqlStatsKnobs
 	knobs.SpanConfig = &spanconfig.TestingKnobs{
 		// Some of these subtests expect multiple (uncoalesced) tenant ranges.
 		StoreDisableCoalesceAdjacent: true,

